add GitHub Actions

szabgab opened the pull request:
As mentioned in the Perl Weekly as well, I think it can be really valuable to enable Continuous Integration (CI) and improve testing on all the CPAN modules where the author is interested in it. Having CI configured will provide fast feedback to the author(s) and will reduce the chances of releasing a module that only works on the computer of the developer. Reducing the manual work-load for the CPAN Tester. Currently GitHub Actions is the most natural CI system for GitHub-based projects. That's why I am sending this PR.

I've enabled several versions of Perl on 3 different platforms. Except 5.36 on Windows as it does not seem to exist.



I have written several blog posts and recorded videos on the subject. You can find them here: https://perlmaven.com/os
If you need further help with the CI system, check out my posts https://perlmaven.com/ci and feel free to ask me.

If you'd like to support my efforts there are several ways to do so https://szabgab.com/support.html

szabgab committed:
add GitHub Actions

bbrtj committed:
Adjust CI github workflow

bbrtj commented:
@szabgab thank you!

I have added some adjustments. Could you double-check that:
- tests are also run on 5.10 (minimum perl version)
- tests are run on schedule only on sundays
- Dist::Zilla is not installed
- dependencies are installed from the `cpanfile`
- tests are run using `prove`, not `dzil test`

Some checks are failing after my changes, so I might have screwed something up.

szabgab commented:
Looks good, but probably you need to add `-l` to prove and maybe also `-v`

bbrtj commented:
`-l` should be taken from `.proverc` (seems like on windows 5.10 it isn't). `-v` is a good idea.

bbrtj committed:
Github action: minor additions

bbrtj committed:
Github action: don't test installed dependencies

bbrtj committed:
Github action: install Test::More on 5.10

bbrtj committed:
Github action: change scheduled run hour

I wake up early...

bbrtj committed:
Github action: only check all versions on Linux

Windows / Mac only checks 5.30 to ensure it works on those archs. Full tests are performed on ubuntu

bbrtj commented:
This looks very good right now, but it still fails on Windows. Seems like `.proverc` is not respected. Do you happen to know the fix for that? Wrong workdir? Maybe App::Prove needs updating?

szabgab commented:
No idea. Have you tried adding the parameter on the command line of `prove` to see if at least that works?

bbrtj committed:
Update ci.yml

bbrtj commented:
Thank you, will try to cope with Windows from here.

szabgab commented:
Excellent!
